Amazon.QLDB.Driver/retry/RetryHandler.cs

```diff
@@ -48,8 +52,9 @@ public T RetriableExecute<T>(Func<T> func, RetryPolicy retryPolicy, Action recov
         {
             Exception last = null;
             int retryAttempt = 1;
+            int recoverRetryAttempt = 1;
 
-            while (retryAttempt <= retryPolicy.MaxRetries + 1)
+            while (retryAttempt <= retryPolicy.MaxRetries + 1 && recoverRetryAttempt - 1 <= this.recoverRetryLimit)
```

Comment:
Why don't we separate the retryPolicy.MaxRetries and recoverRetryAttempts so that if the retry limit is reached for one of them, the other can still retry

Look at Go: https://github.com/awslabs/amazon-qldb-driver-go/blob/master/qldbdriver/qldbdriver.go#L115-L133